### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by megengels** [archlinter/archlint#23]
**Describe the bug**
Multiple detectors generate duplicate smell IDs, causing `snapshot` and `diff` commands to fail. The issue occurs because IDs are generated using function names, and when functions are anonymous (`<anonymous>`), all occurrences in the same file get the same ID.

Additionally, the `SideEffectImport` detector produces false positives on dynamic `import()` expressions (standard Vue Router lazy-loading pattern).

**To Reproduce**
1. Run `pnpm dlx @archlinter/cli snapshot -o .archlint-baseline.json`
2. Run `pnpm dlx @archlinter/cli diff .archlint-baseline.json`
3. See error:
```
error: Snapshot error: Duplicate smell ID: sideeffectimport:ed3eadd8
```
or
```
error: Snapshot error: Duplicate smell ID: cmplx:src/composables/technical-validation/useTechnicalValidationSources.ts:<anonymous>
```
or
```
error: Snapshot error: Duplicate smell ID: nest:src/composables/useNodeExecutionStates.ts:Function '<anonymous>' is too deeply nested
```

**Expected behavior**
1. Each smell should have a unique ID (include file path + line number, not just function name)
2. Dynamic `import()` expressions should not be flagged as side-effect imports

**Screenshots/Output**
```bash
$ grep "sideeffectimport:ed3eadd8" .archlint-baseline.json
      "id": "sideeffectimport:ed3eadd8",
      "id": "sideeffectimport:ed3eadd8",
      "id": "sideeffectimport:ed3eadd8",
      # ... repeated 14 times for different files
```

Example of false positive - standard Vue Router lazy loading:
```typescript
// src/router/index.ts
const router = createRouter({
  routes: [
    {
      path: '/',
      component: () => import('@/pages/HomePage.vue'),  // flagged as side-effect import
    },
  ],
});
```

**Desktop (please complete the following information):**
- OS: Windows 11 (Git Bash / MINGW64)
- archlint version: 0.10.0

**Additional context**
Project: Vue 3 + TypeScript + Vite

Affected detectors:
- `SideEffectImport` - duplicate IDs + false positives on dynamic imports
- `DeepNesting` - duplicate IDs for anonymous functions
- `Complexity` - duplicate IDs for anonymous functions
